api4/channel.go

```diff
@@ -1467,8 +1467,8 @@ func updateChannelMemberSchemeRoles(c *Context, w http.ResponseWriter, r *http.R
 		return
 	}
 
-	schemeRoles := model.SchemeRolesFromJson(r.Body)
-	if schemeRoles == nil {
+	var schemeRoles model.SchemeRoles
+	if jsonErr := json.NewDecoder(r.Body).Decode(&schemeRoles); jsonErr != nil {
 		c.SetInvalidParam("scheme_roles")
```

Comment:
Out of scope of this PR: It would be immensely helpful for API users if the decode errors would also return the json error. Separate ticket?

Reply:
Agreed. Created [MM-37920](https://mattermost.atlassian.net/browse/MM-37920).